### PR TITLE
Enable and search for find command

### LINKED ISSUE #145: Enable FindCommand to support AND operations
Example: `find t/student & t/online class` should return all persons who have both the "student" and "online class" tags.

### PR DESCRIPTION
**Pull request by limyifei1513**
### Find Command Behavior

- **Example 1:**  
  `find m/and t/student n/august`  
  Returns all persons who have **both** the `"student"` tag and `"august"` name.

- **Default Behavior:**  
  Users can omit `m/`
  This will perform **OR** operations by default.

- **Mode (`m/`) Validation:**  
  - Only `AND` or `OR` (case-insensitive) are allowed.  
  - Trailing spaces after `m/and` are acceptable.  
  - Using any other value should show an error.  

- **Example 2:**  
  `find m/and n/ali n/lim`  
  Returns persons whose **name field contains both** `"ali"` and `"lim"`.

- **Error Handling for Other Commands:**  
  Using `m/` in commands other than `FindCommand` should show an error.
  - **Note:** If user inputs `add n/ali a/Block m/11, Kent Ridge Road`, it **will** trigger an error.

#### All main test cases above have been added. If there are any corner cases not covered, please let me know, thanks~

close #145

### CODE REVIEW

File: src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java [hunk 1]
```diff
@@ -53,22 +53,37 @@ public boolean test(Person person) {
...
+                ? phoneKeywords.stream().allMatch(phone::contains)
+                : phoneKeywords.stream().anyMatch(phone::contains);
 
-        boolean matchesTag = tagKeywords.stream()
-                .anyMatch(keyword -> person.getTags().stream()
+        boolean matchesTag = isAndMode
+                ? tagKeywords.stream().allMatch(keyword -> person.getTags().stream()
+                        .anyMatch(tag -> tag.tagName.toLowerCase().contains(keyword.toLowerCase())))
+                : tagKeywords.stream().anyMatch(keyword -> person.getTags().stream()
                         .anyMatch(tag -> tag.tagName.toLowerCase().contains(keyword.toLowerCase())));
```
[the remark this answers is not in the document]
Comment (limyifei1513) [reply src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java]: yes!

**Review by microaugust — COMMENTED** [review]
A few nit pick, but not necessary.

File: src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -63,25 +63,36 @@ public boolean test(Person person) {
...
+        boolean matchesRemark = matchesKeywords(remarkKeywords, remark::contains, isAndMode);
 
-        // Currently only OR semantics are supported.
-        return matchesName || matchesAddress || matchesPhone || matchesTag || matchesRemark;
+        if (matchWord == MatchMode.OR) {
+            return matchesName || matchesAddress || matchesPhone || matchesTag || matchesRemark;
+        } else if (matchWord == MatchMode.AND) {
+            return matchesName && matchesAddress && matchesPhone && matchesTag && matchesRemark;
+        } else {
+            throw new AssertionError("Unhandled match mode: " + matchWord);
```
[the remark this answers is not in the document]
Comment (limyifei1513) [reply src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java]: image
Actually, since MatchMode only has AND or OR, any other value would already trigger an error somewhere else. So I think we don't need to throw the AssertionError? And I think using a switch case make our codes look cleaner right?

File: src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java [hunk 3]
```diff
@@ -62,27 +62,36 @@ public boolean test(Person person) {
...
+        case AND -> matchesName && matchesAddress && matchesPhone && matchesTag && matchesRemark;
+        };
+    }
 
-        // Currently only OR semantics are supported.
-        return matchesName || matchesAddress || matchesPhone || matchesTag || matchesRemark;
+    private boolean matchesKeywords(List<String> keywords, Predicate<String> matcher, boolean isAndMode) {
+        return isAndMode
+                ? keywords.stream().allMatch(matcher)
+                : keywords.stream().anyMatch(matcher);
```
Comment (chiabingxuan) [inline src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java]: This looks really nice :D

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 4]
```diff
@@ -59,6 +63,24 @@ public FindCommand parse(String args) throws ParseException {
...
         List<String> remarkKeywords = getSanitizedKeywords(argMultimap.getAllValues(PREFIX_REMARK));
+        List<String> rawModeKeywords = argMultimap.getAllValues(PREFIX_MODE);
+        List<String> modeKeywords = getSanitizedKeywords(rawModeKeywords);
+
+        MatchMode modeKeyword;
+        // By default, modeKeyword is OR
+        if (rawModeKeywords.isEmpty()) {
+            modeKeyword = MatchMode.OR;
+        } else if (rawModeKeywords.size() > 1) {
+            throw new ParseException(String.format(MESSAGE_MORE_THAN_ONE_MODE, FindCommand.MESSAGE_USAGE));
```
Comment (chiabingxuan) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java]: Image

Looks like there is an existing method to check for duplicate prefix (taken from `AddCommandParser`). Maybe can use this?

File: src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (chiabingxuan) [reply src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java]: I think assertions can still be used to check, in case the error wasn't triggered because of some coding issue
